presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/StreamPropertyDerivations.java

```diff
@@ -601,9 +601,7 @@ private StreamProperties unordered(boolean unordered)
                         exactColumnOrder,
                         partitioningColumns,
                         false,
-                        ActualProperties.builderFrom(otherActualProperties)
-                                .unordered(true)
-                                .build());
+                        otherActualProperties != null ? ActualProperties.builderFrom(otherActualProperties).unordered(true).build() : null);
```

Comment:
is there a way we can write this in a more readable style... maybe assign to a variable